interview-questions.md

```diff
@@ -4,56 +4,76 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destrucuring is the act of simplifying variable names by setting them to this.state where they are originally initialized.
 
-  Researched answer:
+  Researched answer: The object destructuring is a useful JavaScript feature to extract properties from objects and bind them to variables. What's better, object destructuring can extract multiple properties in one statement, can access properties from nested objects, and can set a default value if the property doesn't exist. (https://dmitripavlutin.com/javascript-object-destructuring/#:~:text=The%20object%20destructuring%20is%20a,the%20property%20doesn't%20exist.)
 
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: A DOM event is any event that changes the DOM or the screen where the web page is rendered. For example could be a click or a user entr that changes the pages.
 
-  Researched answer:
+  Researched answer: DOM (Document Object Model) events are actions that occurs as a result of the user action or as result of state change of the elements of a DOM tree. (https://en.wikipedia.org/wiki/DOM_events)
 
 
 
 3. What is the difference between a class and an object?
 
-  Your answer:
+  Your answer: A Class is the blue print for an object that contains standardized behaviors and characteristics. An object is a specific instance of a class. 
 
-  Researched answer:
+  Researched answer: A class is a template for objects. A class defines object properties including a valid range of values, and a default value. A class also describes object behavior. An object is a member or an "instance" of a class. https://www.ncl.ucar.edu/Document/HLUs/User_Guide/classes/classoview.shtml
 
 
 
-4. What is the difference between a HTML div and a span?
 
-  Your answer:
+4. What is the difference between a HTML div and a span?
 
-  Researched answer:
+  Your answer: HTML div is a division that has a defined opening and closing with as many items inside, but a span is just one line that wraps a small portion of text.
 
+  Researched answer: 
 
 
 5. What is React.Fragment (or <>) and why would you want to use it?
 
-  Your answer:
+  Your answer: React.Fragment is a division in a react component that wraps up multiple elements and is necessary to use because react components can only have one return.
+
+  Researched answer: Both the tags (<div> and <span>) are used to represent the part of the webpgage, <div> tag is used a as block part of the webpage and <span> tag is used as a inline part of the webpage (https://www.geeksforgeeks.org/difference-between-div-and-span-tag-in-html/)
```

Comment:
Found that missing researched answer! Looks like it somehow got put in for your react.fragment one.